odom_frame: "map"
base_frame: "odom"

kf_dist_linear: 0.1     # 关键线距离m
kf_dist_angular: 0.1    # 关键角度单位是弧度
kf_scan_count: 5        # 关键扫描次数5

max_angular_correction_deg:  540.0   # 扫描之间的最大角位移
max_linear_correction:  3.0         # 扫描之间最大位移
max_iterations: 10                  # 最大ICP周期迭代次数
epsilon_xy: 0.000001                # εxy
epsilon_theta: 0.000001             # εtheta
max_correspondence_dist: 1.0        # 点云匹配中两个点之间的最大距离 过小导致匹配的点数过少，过大匹配错误的点
sigma: 0.010                        # sigma参数通常是在点云匹配的过程中用于计算距离权重的。可以根据具体的应用场景来选择合适的sigma值，以提高匹配的准确性和速度

# CSM库中的参数设定
# use_corr_tricks是一个bool类型的参数，用于控制点云匹配过程中是否使用一些技巧来提高匹配的准确性和速度。
use_corr_tricks: 1                  # 对于大规模的数据集和对实时性要求较高的应用场景，可以将use_corr_tricks设置为true，以获得更好的性能。

# 当restart为true时，C++CSM库会在每次迭代中重新启动匹配算法。这样做的好处是可以避免算法陷入局部最优解，从而提高匹配的准确性。但同时也会增加算法的计算时间，降低匹配的效率。
restart: 0                          # 默认是false 需要设置为true

# 用于控制点云匹配过程中重新启动算法的条件参数的值为0.01。如果需要改变这个值，可以将其设置为其他的数值，以获得更好的匹配效果。
# 算法的平均误差超过restart_threshold_mean_error时，C++CSM库会重新启动匹配算法。这个平均误差是指匹配过程中所有对应点之间的距离的平均值。
restart_threshold_mean_error: 0.01  # 如果需要高精度的匹配结果，则可以将其设置的较小；如果对匹配速度要求较高，则可以将其设置的较大。

# 算法的迭代时间超过restart_dt时，C++CSM库会重新启动匹配算法。这个迭代时间是指匹配算法在一次迭代中所需要的时间
restart_dt: 1.0                     # 默认情况下，restart_dt参数的值为0.01。如果需要改变这个值，可以将其设置为其他的数值，以获得更好的匹配效果。
                                    # 如果需要高精度的匹配结果，则可以将其设置的较小；如果对匹配速度要求较高，则可以将其设置的较大。
# 算法的旋转角度超过restart_dtheta时，C++CSM库会重新启动匹配算法。这个旋转角度是指匹配过程中两个点云之间的旋转角度。
restart_dtheta: 0.1                 # 默认情况下，restart_dtheta参数的值为0.1  如果需要高精度的匹配结果，则可以将其设置的较小；如果对匹配速度要求较高，则可以将其设置的较大。

# C++CSM库使用点云聚类的方法将点云中的点分为不同的簇。在聚类的过程中，如果两个点之间的距离小于clustering_threshold，则它们被认为是同一个簇中的点。如果它们之间的距离大于clustering_threshold，则它们被认为是不同簇中的点。
clustering_threshold: 0.25          # clustering_threshold参数的值为0.05 如果需要将点云分为更多的簇，则可以将其设置的较小；如果需要将点云分为较少的簇，则可以将其设置的较大

# C++CSM库在点云匹配的过程中，需要计算每个点的法向量。为了提高法向量的准确性，C++CSM库会在每个点周围的邻域中计算法向量。orientation_neighbourhood参数指定了这个邻域的大小，即在这个半径内的点会被用来计算法向量。
orientation_neighbourhood: 20       # orientation_neighbourhood参数的值为0.2 如果需要更准确的法向量计算结果，则可以将其设置的较小；如果对计算速度要求较高，则可以将其设置的较大。

# C++CSM库在点云匹配的过程中，可以使用点到点距离或点到线距离来计算点云之间的差异。如果use_point_to_line_distance为true，则使用点到线距离；否则，使用点到点距离。
# 点到线距离可以更准确地描述点云之间的差异，因为它考虑了点云之间的旋转和平移。但同时，点到线距离的计算也比较复杂，需要进行额外的计算。因此，在实际应用中，可以根据具体的应用场景来选择合适的计算方式。
use_point_to_line_distance: 1       # use_point_to_line_distance参数的值为false，即使用点到点距离计算点云之间的差异。如果需要改变这个值，可以将其设置为true，以使用点到线距离计算点云之间的差异

# 用于控制点云匹配过程中是否使用alpha测试 使用alpha测试可以去除一些被遮挡的点，从而提高匹配的准确性。但同时也会增加计算量，降低匹配的效率
do_alpha_test: 0                    # 如果需要更准确的匹配结果，则可以使用alpha测试；如果对匹配速度要求较高，则可以不使用alpha测试。

# 可以使用alpha测试来删除一些不可见的点。当点云中某个点的法向量与摄像机之间的夹角大于do_alpha_test_thresholdDeg时，则认为该点是可见的，否则认为该点被遮挡了。
# 如果需要更准确的匹配结果，则可以将其设置的较小；如果对匹配速度要求较高，则可以将其设置的较大
do_alpha_test_thresholdDeg: 20.0    # 默认情况下，do_alpha_test_thresholdDeg参数的值为90度，即所有点都会被认为是可见的。如果需要改变这个值，可以将其设置为其他的数值，以获得更好的alpha测试效果

# 用于控制点云匹配过程中的离群点数量
# outliers_maxPerc参数指定了最大的离群点数量占点云总数的比例。如果点云中离群点的数量超过了这个比例，则这些点将被认为是离群点，并被排除在匹配的过程之外
# 默认情况下，outliers_maxPerc参数的值为0.9，即最多允许90%的点被认为是离群点
outliers_maxPerc: 0.90              # 如果需要更准确的匹配结果，则可以将其设置的较小；如果对匹配速度要求较高，则可以将其设置的较大

# outliers_adaptive_order参数用于控制离群点数量的自适应性，即根据点云的特征自动调整离群点数量的阈值
# 如果点云的特征较为复杂，可以考虑启用自适应性；如果点云的特征比较简单，则可以不使用自适应性
outliers_adaptive_order: 0.7        # 默认情况下，outliers_adaptive_order参数的值为false，即离群点数量的阈值不会根据点云的特征自适应调整

# 用于控制点云匹配过程中离群点数量的自适应性，根据点云的特征自动调整离群点数量的阈值
# 默认情况下，outliers_adaptive_mult参数的值为0.04，即离群点数量阈值的初始值为4%。如果需要改变这个值，可以将其设置为其他的数值，以获得更好的自适应性能
outliers_adaptive_mult: 2.0         # 如果需要更准确的匹配结果，则可以将其设置的较小；如果对匹配速度要求较高，则可以将其设置的较大

# 用于控制点云匹配过程中是否使用可见性测试，使用可见性测试可以去除一些被遮挡的点，从而提高匹配的准确性。但同时也会增加计算量，降低匹配的效率
do_visibility_test: 0               # 如果需要更准确的匹配结果，则可以使用可见性测试；如果对匹配速度要求较高，则可以不使用可见性测试。

# C++CSM库在点云匹配的过程中，可能会存在一些重复的点，即空间位置相同但其他属性不同的点。outliers_remove_doubles参数用于控制是否删除这些重复的点
outliers_remove_doubles: 0          # 如果点云中存在大量的重复点，则可以考虑删除重复的点，以减小点云的规模；如果不需要考虑点云的规模，或者点云中的重复点比较少，则可以不删除重复的点。

# 可以使用调试模式来检查代码的正确性。如果debug_verify_tricks为true，则使用调试模式；否则，不使用调试模式
do_compute_covariance: 0            # 如果需要检查代码的正确性，则可以使用调试模式；如果对匹配速度要求较高，则可以不使用调试模式

# 如果use_ml_weights为true，则使用最大似然权重；否则，不使用最大似然权重。使用最大似然权重可以提高点云匹配的精度，但同时也会增加计算量，降低匹配的效率。
use_ml_weights: 0                   # 如果需要更准确的匹配结果，则可以使用最大似然权重；如果对匹配速度要求较高，则可以不使用最大似然权重

# 如果use_sigma_weights为true，则使用标准偏差权重；否则，不使用标准偏差权重。使用标准偏差权重可以提高点云匹配的精度，但同时也会增加计算量，降低匹配的效率
use_sigma_weights: 0                # 如果需要更准确的匹配结果，则可以使用标准偏差权重；如果对匹配速度要求较高，则可以不使用标准偏差权重
